packages/dbml-connector/src/connectors/mssqlConnector.ts

```diff
@@ -49,19 +49,51 @@ const getValidatedClient = async (connection: string): Promise<sql.ConnectionPoo
 
 const convertQueryBoolean = (val: string | null): boolean => val === 'YES';
 
-const getFieldType = (data_type: string, default_type: DefaultType, character_maximum_length: number, numeric_precision: number, numeric_scale: number): string => {
+const getFieldType = (data_type: string, character_maximum_length: number, numeric_precision: number, numeric_scale: number): string => {
   if (MSSQL_DATE_TYPES.includes(data_type)) {
     return data_type;
   }
+
+  // timestamp is a synonym of rowversion
+  // microsoft recommends that we should not use timestamp, so we will convert it back to rowversion
+  // https://learn.microsoft.com/en-us/sql/t-sql/data-types/rowversion-transact-sql?view=sql-server-ver15
+  if (data_type === 'timestamp') {
+    return 'rowversion';
+  }
+
+  // process numeric -based type
   if (data_type === 'bit') {
     return data_type;
   }
-  if (numeric_precision && numeric_scale && default_type === 'number') {
-    return `${data_type}(${numeric_precision},${numeric_scale})`;
+  // if precision != 0 => numeric-based column
+  if (numeric_precision) {
+    return numeric_scale
+      ? `${data_type}(${numeric_precision},${numeric_scale})`
+      : `${data_type}(${numeric_precision})`;
   }
```

Comment:
How about date/time data type em?

Reply:
Let me check again a, I forgot it

Reply:
For date/time, datetime2 and datetimeoffset support custom fractional seconds precision, see https://learn.microsoft.com/en-us/sql/t-sql/data-types/datetime2-transact-sql?view=sql-server-ver15

So I fetch the numeric scale for them.

Fixed in commit https://github.com/holistics/dbml/pull/645/commits/0b972fc75d2280aff819329ee5293c698b361379